fe/fe-core/src/main/java/com/starrocks/qe/scheduler/slot/Slot.java

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.qe.scheduler.slot;
+
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TResourceSlot;
+import com.starrocks.thrift.TUniqueId;
+
+public class Slot {
+    public static final long ABSENT_GROUP_ID = -1;
+
+    private final TUniqueId slotId;
+    private final TNetworkAddress requestEndpoint;
+
+    private final long groupId;
+
+    private final int numSlots;
```

Comment:
Could one slot have multi slots?

Reply:
Yes, a logical slot may have multiple physical slots. For now, a logical slot only has one physical slot.